### ISSUE TITLE
Error: Plugin did not respond

### ISSUE THREAD

**Issue by jckeme-rs** [Azure/terraform-provider-azapi#112]
Hi, 

I'm attempting to update a slot configuration with the Azure Api referenced here:
https://docs.microsoft.com/en-us/rest/api/appservice/web-apps/update-slot-configuration-names

This feature isn't fully supported using the Azurerm Provider.

This Azure API supports a PUT method for which I'm attempting to use azapi_update_resource. 

The resource block (within a module) looks like this:

```
resource "azapi_update_resource" "slot_update" {
  for_each = local.sticky_slot_settings
  type      = "Microsoft.Web/sites/config@2021-03-01"

  parent_id = azurerm_app_service.app[0].id 

  name = "slotConfigNames"
  body = jsonencode({
    properties = {
      "connectionStringNames": [],
      "appSettingNames": [ "${each.key}" ],
      "azureStorageConfigNames": []
    }
  })

  ignore_missing_property = true
  ignore_casing = true

}
```
Terraform plan looks fine, however the azapi plugin seems to crash on apply. 

```
The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.

Stack trace from the terraform-provider-azapi_v0.2.0.exe plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 53 [running]:
github.com/Azure/terraform-provider-azapi/utils.GetId(...)
        github.com/Azure/terraform-provider-azapi/utils/helper.go:17
github.com/Azure/terraform-provider-azapi/internal/services.resourceAzApiUpdateResourceCreateUpdate(0x14a6c2d0, {0xa96780, 0x14c74af8})
        github.com/Azure/terraform-provider-azapi/internal/services/azapi_update_resource.go:161 +0xa8b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x14a8caf0, {0x333a10c, 0x14856c30}, 0x14a6c2d0, {0xa96780, 0x14c74af8})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.8.0/helper/schema/resource.go:318 +0x166
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14a8caf0, {0x333a10c, 0x14856c30}, 0x14dee230, 0x14a6c190, {0xa96780, 0x14c74af8})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.8.0/helper/schema/resource.go:456 +0x876
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x14a800a0, {0x333a10c, 0x14856c30}, 0x14856db0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.8.0/helper/schema/grpc_provider.go:977 +0xeb4
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x14a86f78, {0x333a16c, 0x1488a4c8}, 0x14958040)
        github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/tf5server/server.go:332 +0x8e
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xb21540, 0x14a86f78}, {0x333a16c, 0x1488a4c8}, 0x14856a50, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x18a
google.golang.org/grpc.(*Server).processUnaryRPC(0x14b22120, {0x3348014, 0x14b9e100}, 0x148ca280, 0x14a86f90, 0x378f9e0, 0x0)
        google.golang.org/grpc@v1.39.0/server.go:1292 +0xd11
google.golang.org/grpc.(*Server).handleStream(0x14b22120, {0x3348014, 0x14b9e100}, 0x148ca280, 0x0)
        google.golang.org/grpc@v1.39.0/server.go:1617 +0xa2d
google.golang.org/grpc.(*Server).serveStreams.func1.2(0x14b88560, 0x14b22120, {0x3348014, 0x14b9e100}, 0x148ca280)
        google.golang.org/grpc@v1.39.0/server.go:940 +0x92
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.39.0/server.go:938 +0x1a7

Error: The terraform-provider-azapi_v0.2.0.exe plugin crashed!
```

Alongside, some Debug logs indicate entries such as:

```
2022-05-18T11:21:16.567-0400 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/azure/azapi/0.2.0/windows_386/terraform-provider-azapi_v0.2.0.exe args=[.terraform/providers/registry.terraform.io/azure/azapi/0.2.0/windows_386/terraform-provider-azapi_v0.2.0.exe]
2022-05-18T11:21:16.572-0400 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/azure/azapi/0.2.0/windows_386/terraform-provider-azapi_v0.2.0.exe pid=29816
2022-05-18T11:21:16.572-0400 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/azure/azapi/0.2.0/windows_386/terraform-provider-azapi_v0.2.0.exe
2022-05-18T11:21:16.605-0400 [INFO]  provider.terraform-provider-azapi_v0.2.0.exe: configuring server automatic mTLS: timestamp=2022-05-18T11:21:16.604-0400
2022-05-18T11:21:16.658-0400 [DEBUG] provider.terraform-provider-azapi_v0.2.0.exe: plugin address: address=127.0.0.1:10001 network=tcp timestamp=2022-05-18T11:21:16.658-0400
2022-05-18T11:21:16.658-0400 [DEBUG] provider: using plugin: version=5
2022-05-18T11:21:16.738-0400 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2022-05-18T11:21:16.752-0400 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/azure/azapi/0.2.0/windows_386/terraform-provider-azapi_v0.2.0.exe pid=29816
2022-05-18T11:21:16.752-0400 [DEBUG] provider: plugin exited
2022-05-18T11:21:16.753-0400 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2022-05-18T11:21:16.768-0400 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/azurerm/2.99.0/windows_386/terraform-provider-azurerm_v2.99.0_x5.exe pid=23412
2022-05-18T11:21:16.768-0400 [DEBUG] provider: plugin exited
```

Could this be something wrong with the plugin when trying to interact with this API endpoint or perhaps something incorrect with the configuration?

**Comment by ms-henglu** [comment]
Hi @jckeme-rs ,

Thank you for taking time to report this issue!

The reason is that `slotConfigNames` is not a valid config name for "Microsoft.Web/sites/config@2021-03-01", the allowed configs are listed in the followings.
[authsettings](https://docs.microsoft.com/en-us/azure/templates/microsoft.web/sites/config-authsettings)
[authsettingsV2](https://docs.microsoft.com/en-us/azure/templates/microsoft.web/sites/config-authsettingsv2)
[azurestorageaccounts](https://docs.microsoft.com/en-us/azure/templates/microsoft.web/sites/config-azurestorageaccounts)
[backup](https://docs.microsoft.com/en-us/azure/templates/microsoft.web/sites/config-backup)
[connectionstrings](https://docs.microsoft.com/en-us/azure/templates/microsoft.web/sites/config-connectionstrings)
[logs](https://docs.microsoft.com/en-us/azure/templates/microsoft.web/sites/config-logs)
[metadata](https://docs.microsoft.com/en-us/azure/templates/microsoft.web/sites/config-metadata)
[pushsettings](https://docs.microsoft.com/en-us/azure/templates/microsoft.web/sites/config-pushsettings)
[slotConfigNames](https://docs.microsoft.com/en-us/azure/templates/microsoft.web/sites/config-slotconfignames)
[web](https://docs.microsoft.com/en-us/azure/templates/microsoft.web/sites/config-web)

But there's also a bug in the provider, it should not crash, I'll fix it.

Thanks!

**Comment by ms-henglu** [comment]
Hi @jckeme-rs ,

Thank you for taking time to report this issue!

It seems a server-side bug that when GET this config, its id is null.
```
GET https://management.azure.com/subscriptions/0000-000-000-0000/resourceGroups/henglu5192/providers/Microsoft.Web/sites/uniwebndhjate7835/config/slotConfigNames?api-version=2022-03-01

{
    "id": null,
    "name": "uniwebndhjate7835",
    "type": "Microsoft.Web/sites",
    "location": "East US",
    "tags": {
        "displayName": "PartsUnlimitedWebsite",
        "hidden-related:/subscriptions/0000-000-000-0000/resourceGroups/henglu5192/providers/Microsoft.Web/serverfarms/MHCHostPlanN14w3257": "Resource"
    },
    "properties": {
        "connectionStringNames": [
            "DefaultConnectionString"
        ],
        "appSettingNames": [
            "keys:ApplicationInsights:InstrumentationKey"
        ],
        "azureStorageConfigNames": [],
        "secretsCollectionNames": []
    }
}
```

I'll contact service team to fix it. But there's also a bug in the provider, it should not crash, I'll fix it.

Thanks!

**Comment by ms-henglu** [comment]
link https://github.com/Azure/azure-rest-api-specs

**Comment by ms-henglu** [comment]
Hi @jckeme-rs , 

I've confirmed that has been fixed in https://github.com/Azure/terraform-provider-azapi/releases/tag/v0.2.1, would you please give it a try?

**Comment by jckeme-rs** [comment]
Hi @ms-henglu , 

I can confirm that this is now fixed in the 0.2.1 release. 

Thank you for the effort!
